GW2EIEvtcParser/EIData/ProfHelpers/Engineer/MechanistHelper.cs

```diff
@@ -20,6 +21,19 @@ internal static class MechanistHelper
             new EffectCastFinderByDst(OverclockSignetSkill, EffectGUIDs.MechanistOverclockSignet).UsingDstSpecChecker(Spec.Mechanist),
 
             // Mech
+            new DamageCastFinder(CrashDown, CrashDown).UsingDisableWithEffectData(),
+            new MinionSpawnCastFinder(CrashDown, (int)MinionID.JadeMech)
+                .UsingChecker((spawn, combatData, agentData, skillData) =>
+                {
+                    // TODO: what if shift signet ports the minion away from effect before arc polls the position? so far unable to produce
+                    Point3D pos = combatData.GetMovementData(spawn.Src).OfType<PositionEvent>().FirstOrDefault(evt => evt.Time + ServerDelayConstant >= spawn.Time)?.ToPoint();
+                    if (pos != null && combatData.TryGetEffectEventsByGUID(EffectGUIDs.MechanistCrashDownImpact, out IReadOnlyList<EffectEvent> effects))
+                    {
+                        return FindRelatedEvents(effects, spawn.Time + 800).Any(effect => pos.Distance2DToPoint(effect.Position) < 10.0f);
```

Comment:
Should be changed to Count(...) == 1 instead of Any(...) for the very rare situations where two crash downs happen  at the same time on the same place.
However, wasn't Crash Down an animated skill for the Jade Mech?

Reply:
I think if two Mech Crash Downs at the same place `Any(...)` should work fine? `Count(...) == 1` would mean neither of the two Crash Down casts get detected in that case. The scenario where `Any(...)` does not work correctly is when Mech A crashes down at the location and Mech B spawns there without Crash Down. Not sure if that is a realistic thing to happen? To be honest I don't even know under what circumstances you could get a spawn event for a Mech without Crash Down.

Crash Down clearly plays an animation ingame, but it seems Arc does not detect it...? Looking at logs the first cast from the Mechs spawning via Crash Down is usually some weapon drawing ~3s later.